File: homework04/api_models.py
from typing import Optional, Dict, List

from pydantic import BaseModel


class BaseUser(BaseModel):
    """ Модель пользователя с базовыми полями """
    id: int
    first_name: str
    last_name: str
    online: int
    deactivated: Optional[str]


class User(BaseUser):
    """ Модель пользователя с необязательным полем дата рождения """
    bdate: Optional[str]


class Message(BaseModel):
    """ Модель сообщения """
    date: int
    from_id: int
    id: int
    out: int
    peer_id: int
    text: str
    conversation_message_id: int
    fwd_messages: Optional[List]
    important: bool
    random_id: int
    attachments: Optional[List]
    is_hidden: bool
    reply_message: Optional[Dict]

File: homework04-go/apiModels.go
package main

type User struct {
	Id          int
	First_name  string
	Last_name   string
	Online      int
	Deactivated string
	Bdate       string
}

type Message struct {
	date   int
	fromId int
	id     int
	out    int
	peerId int
	text   string
}

type Response struct {
	Items []User
	Count float64
}

type FriendsResponse struct {
	Response Response
}
